pulsar-client-admin/src/main/java/org/apache/pulsar/client/admin/internal/PulsarAdminBuilderImpl.java

```diff
@@ -54,13 +55,23 @@ private PulsarAdminBuilderImpl(ClientConfigurationData conf) {
 
     @Override
     public PulsarAdminBuilder clone() {
-        return new PulsarAdminBuilderImpl(conf.clone());
+        PulsarAdminBuilderImpl pulsarAdminBuilder = new PulsarAdminBuilderImpl(conf.clone());
+        pulsarAdminBuilder.clientBuilderClassLoader = clientBuilderClassLoader;
+        pulsarAdminBuilder.gzipCompressionEnabled = gzipCompressionEnabled;
+        return pulsarAdminBuilder;
     }
 
     @Override
     public PulsarAdminBuilder loadConf(Map<String, Object> config) {
         conf = ConfigurationDataUtils.loadData(config, conf, ClientConfigurationData.class);
         setAuthenticationFromPropsIfAvailable(conf);
+        if (config.containsKey("gzipCompressionEnabled")) {
+            if (config.get("gzipCompressionEnabled") instanceof Boolean) {
```

Comment:
I would personally extract it once `config.get("gzipCompressionEnabled")`

Reply:
I agree, that would make sense. This PR is still in draft mode.

Reply:
Fixed